### PR TITLE
Update dependencies and use modules from roc-lang/roc

### PR DESCRIPTION
**Pull request by lukewilliamboswell**
- `cargo update` 
- Use `roc_std` and `roc_std_heap` directly from roc-lang/roc to prevent going out of sync

### CODE REVIEW

File: crates/roc_host/Cargo.toml
Comment (lukewilliamboswell) [inline crates/roc_host/Cargo.toml:1]: I wasn't 100% on why these were locked using `=`, I removed to see if `cargo update` would upgrade these versions but it doesn't seem to have changed anything.

Comment (Anton-4) [reply crates/roc_host/Cargo.toml:1]: Rust recommends this for applications (as opposed to rust packages depended on by others). I think we can think of the basic-cli platform as an application in this regard. I think I added this due to a tricky bug caused by hyper-rustls where nothing changed in the basic-cli code base but suddenly I could not built it anymore due to cargo silently upgrading to the next patch release.

File: crates/roc_host/src/lib.rs [hunk 1]
```diff
@@ -16,12 +16,6 @@ use std::time::{Duration, SystemTime, UNIX_EPOCH};
 use std::{env, io};
 use tokio::runtime::Runtime;
 
-/// Implementation of the host.
-/// The host contains code that calls the Roc main function and provides the
-/// Roc app with functions to allocate memory and execute effects such as
-/// writing to stdio or making HTTP requests.
```
Comment (Anton-4) [inline crates/roc_host/src/lib.rs:22]: This comment was for the entire file, not just for `mod heap`. Perhaps rustfmt removed the lines in between the comments and the code...

File: build.roc [hunk 2]
```diff
@@ -42,8 +42,6 @@ run = \{ debugMode, maybeRoc } ->
 
     rocVersion! rocCmd
 
-    generateGlue! rocCmd
-
```
Comment (Anton-4) [inline build.roc:46]: I don't completely understand why this is being removed now, can you explain?

**Review by Anton-4 — CHANGES_REQUESTED** [review]

File: build.roc (hunk 2 above)
Comment (lukewilliamboswell) [reply build.roc:46]: Yes, the "glue" only provide roc_std, and I think using cargo to get this directly from roc-lang/roc is a much better method for now. In future we may introduce glue back into the platform, but we have replaced all the interface types with hand-rolled implementations to be sure we don't have bugs lurking from RustGlue's current implementation.

File: crates/roc_host/Cargo.toml
Comment (lukewilliamboswell) [reply crates/roc_host/Cargo.toml:1]: No worries, added back in 988df0200db7a41b61cd70770fad606b653c338a

File: crates/roc_host/src/lib.rs (hunk 1 above)
Comment (lukewilliamboswell) [reply crates/roc_host/src/lib.rs:22]: add in a983b2e0976521f8d43053b2ffc35a8fe69fab45, also checked agains the `cargo doc --open`

Screenshot 2024-09-29 at 09 35 41

File: crates/roc_host/src/lib.rs [hunk 3]
```diff
@@ -1,10 +1,15 @@
+//! Implementation of the host.
+//! The host contains code that calls the Roc main function and provides the
+//! Roc app with functions to allocate memory and execute effects such as
+//! writing to stdio or making HTTP requests.
```
Comment (Anton-4) [inline crates/roc_host/src/lib.rs:4]: Good to know that `//!` is the magic syntax :)

**Review by Anton-4 — APPROVED** [review]
Thanks @lukewilliamboswell :)

**Pull request merged**